src/genn/backends/opencl/backend.cc

```diff
@@ -892,7 +892,7 @@ MemAlloc Backend::genVariableAllocation(CodeStream& os, const std::string& type,
 
 	// If variable is present on device then initialize the device buffer
 	if (loc & VarLocation::DEVICE) {
-		os << getVarPrefix() << name << " = cl::Buffer(clContext, CL_MEM_READ_WRITE, " << count << " * sizeof (" << type << "), " << name << ");" << std::endl;
+		os << getVarPrefix() << name << " = cl::Buffer(clContext, CL_MEM_READ_WRITE, CL_MEM_COPY_HOST_PTR " << count << " * sizeof (" << type << "), " << name << ");" << std::endl;
```

Comment:
this seems broken - you're putting ``CL_MEM_COPY_HOST_PTR`` into the size

Reply:
Sorry about it @neworderofjamie 
I missed this. Looks like @9inpachi missed this too.

Reply:
OH.
I surely did miss it.
Pushed the fixes now.